### ISSUE TITLE
Test failures from stats and lambdify in master

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#20292]
The stats tests fail on current master. Possibly this depends on the numpy version.

This is a release blocker for 1.7.

```console
$ bin/test sympy/stats/tests/test_continuous_rv.py
============================================================= test process starts =============================================================
executable:         /Users/enojb/current/sympy/38venv/bin/python  (3.8.1-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.8
numpy:              1.18.3
random seed:        893412
hash randomization: on (PYTHONHASHSEED=2425498222)

sympy/stats/tests/test_continuous_rv.py[88] ...w....E...................w............................f.....E..........ww.........Es.     [FAIL]

________________________________________________________________ slowest tests ________________________________________________________________
sympy/stats/tests/test_continuous_rv.py::test_trapezoidal - Took 14.720 seconds
sympy/stats/tests/test_continuous_rv.py::test_exponential - Took 22.614 seconds
sympy/stats/tests/test_continuous_rv.py::test_prefab_sampling - Took 26.954 seconds
_______________________________________________________________________________________________________________________________________________
_______________________________________ sympy/stats/tests/test_continuous_rv.py:test_sample_continuous ________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 332, in test_sample_continuous
    assert next(sample(Z)) in Z.pspace.domain.set
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1673, in _ppf_single
    while self._ppf_to_solve(right, q, *args) < 0.:
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-2>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_1448, 0),less(Dummy_1448, PINF))),True], [exp(-Dummy_1448),0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'
_______________________________________________________________________________________________________________________________________________
________________________________________ sympy/stats/tests/test_continuous_rv.py:test_prefab_sampling _________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 1332, in test_prefab_sampling
    assert next(sample(var)) in var.pspace.domain.set
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1678, in _ppf_single
    return optimize.brentq(self._ppf_to_solve,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/optimize/zeros.py", line 780, in brentq
    r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-216>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_12577, 0),less_equal(Dummy_12577, 1))),True], [1,0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'
_______________________________________________________________________________________________________________________________________________
__________________________________________ sympy/stats/tests/test_continuous_rv.py:test_sample_scipy __________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 1633, in test_sample_scipy
    samps = next(sample(X, size=size, library='scipy'))
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1678, in _ppf_single
    return optimize.brentq(self._ppf_to_solve,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/optimize/zeros.py", line 780, in brentq
    r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-285>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_15389, 0),less_equal(Dummy_15389, 1))),True], [1,0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'

========================== tests finished: 79 passed, 5 skipped, 1 expected to fail, 3 exceptions, in 147.80 seconds ==========================
DO *NOT* COMMIT!
```

**Comment by oscarbenjamin** [comment]
This is coming from the sampling tests @Smit-create @czgdp1807

**Comment by asmeurer** [comment]
I'm guessing this test wasn't properly included in the external dependencies test list on Travis.

**Comment by asmeurer** [comment]
It looks like `logical_and` is a deprecated wrapper from `scipy._lib.deprecation`. 

This can be fixed by making sure `lambdify` uses `['numpy', 'scipy']` instead of just `'scipy'` as the module argument.

```diff
diff --git a/sympy/stats/crv.py b/sympy/stats/crv.py
index f599486dd5..99f625fd2a 100644
--- a/sympy/stats/crv.py
+++ b/sympy/stats/crv.py
@@ -156,7 +156,7 @@ def _sample_scipy(cls, dist, size):
         # scipy does not require map as it can handle using custom distributions
         from scipy.stats import rv_continuous
         z = Dummy('z')
-        handmade_pdf = lambdify(z, dist.pdf(z), 'scipy')
+        handmade_pdf = lambdify(z, dist.pdf(z), ['numpy', 'scipy'])
         class scipy_pdf(rv_continuous):
             def _pdf(self, x):
                 return handmade_pdf(x)
```

(there's also other places that use `'scipy'` that should be changed).

More generally, though, passing `'scipy'` to lambdify should probably do this automatically.

**Comment by czgdp1807** [comment]
For now I have removed, `'scipy'` argument from `lambdify` where it was causing failures. After the release, probably, we can see from where itself such removals should be done.

**Comment by asmeurer** [comment]
I opened https://github.com/sympy/sympy/issues/20294 for changing the way `lambdify(modules='scipy')` works. That would be easy enough to fix, but there is a possibility it could break something, so it's probably better to not try to push it for the release given that fixing stats is easy enough.

**Comment by oscarbenjamin** [comment]
A fix has been merged for both 1.7 and master. There is an issue for further fixes (#20294).

I'm going to close this as fixed.
